taichi/common/serialization.h

```diff
@@ -313,25 +337,36 @@ class BinarySerializer : public Serializer {
     }
   }
 
+  template <typename T>
+  void operator()(const char *, const T &val) {
+    this->process(val);
+  }
```

Comment:
Is this one ever used anywhere? And why doesn't it call `this->process(const char *key, const std::string &val)`?

Reply:
There is no `process(const char* key, T val)` interface. Note that the binary serializers don't need the key :-)